### PR TITLE
Parallelize update tests

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR aims to speed up the update tests by parallelizing them. As a single class, surefire/failsafe cannot parallelize the 30 parameterized test cases using multiple forks. The solution is to split them off into multiple classes, where each class will be executed in a different fork. The shared assertion logic is then extracted to `ContainerStateAssert`, as all the assertions are asserting the current state of the broker.

## Related issues

related to #5873

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by npepinpe** [comment]
Some notes here:

I started a `ContainerStateAssert` to be able to reuse the assertions that were in the previous `UpdateTest` class (e.g. await process is completed, await there's a snapshot, etc.). It seemed like the right choice, since we want to assert some properties on the current container's state. I'm not too sure about the eventual stuff though, e.g. `assertThat(state).eventually().hasSnapshotAvailable(1)`. I thought it was a clever idea, but I'm wary of clever ideas, so I would love for someone to challenge this as unnecessary DSL-like which just obscures the code for minimal "DRY"-ness.

File: update-tests/src/test/java/io/zeebe/test/ContainerState.java [hunk 1]
```diff
@@ -18,13 +18,14 @@
...
 import net.jodah.failsafe.RetryPolicy;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testcontainers.containers.ContainerFetchException;
 import org.testcontainers.containers.ContainerLaunchException;
 import org.testcontainers.containers.Network;
 
-public class ContainerState {
+public class ContainerState implements CloseableResource {
```
Comment (npepinpe) [inline update-tests/src/test/java/io/zeebe/test/ContainerState.java:28]: In junit5, `CloseableResource` objects which are put in a store are automatically closed when the test lifecycle ends (so if it's on a parameter or field, when the test ends; if it's on a static field, when the class/suite ends).

File: update-tests/src/test/java/io/zeebe/test/ContainerStateExtension.java [hunk 2]
```diff
@@ -8,39 +8,52 @@
...
-
-  public ContainerStateExtension(final ContainerState state) {
-    this.state = state;
-  }
+/**
+ * This extension injects a new {@link ContainerState} at runtime into any test which adds a {@link
+ * ContainerState} parameter, and stores it in a {@link Store}. This ensures that the resource is
+ * properly closed (since {@link ContainerState} implements {@link CloseableResource}).
+ *
+ * <p>Note however that it currently only supports injecting a single state, since the stored state
```
Comment (npepinpe) [inline update-tests/src/test/java/io/zeebe/test/ContainerStateExtension.java:26]: Easily extended if we need more by storing a collection. Note that we'd need to make the collection itself a `CloseableResource` to ensure everything is cleaned up.

File: update-tests/src/test/java/io/zeebe/test/ContainerState.java [hunk 3]
```diff
@@ -18,15 +18,16 @@
...
 import org.slf4j.LoggerFactory;
 import org.testcontainers.containers.ContainerFetchException;
 import org.testcontainers.containers.ContainerLaunchException;
 import org.testcontainers.containers.Network;
 
-public class ContainerState {
+final class ContainerState implements CloseableResource {
   private static final RetryPolicy<Void> CONTAINER_START_RETRY_POLICY =
-      new RetryPolicy().withMaxRetries(5).withBackoff(3, 30, ChronoUnit.SECONDS);
+      new RetryPolicy<Void>().withMaxRetries(5).withBackoff(3, 30, ChronoUnit.SECONDS);
```
Comment (npepinpe) [inline update-tests/src/test/java/io/zeebe/test/ContainerState.java:30]: Fixes a warning about type erasure

File: update-tests/src/test/java/io/zeebe/test/ContainerState.java [hunk 4]
```diff
@@ -18,15 +18,16 @@
...
 import net.jodah.failsafe.RetryPolicy;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testcontainers.containers.ContainerFetchException;
 import org.testcontainers.containers.ContainerLaunchException;
 import org.testcontainers.containers.Network;
 
-public class ContainerState {
+final class ContainerState implements CloseableResource {
```
Comment (npepinpe) [inline update-tests/src/test/java/io/zeebe/test/ContainerState.java:28]: In junit5, `CloseableResource` instances which are stored in an extension store will be automatically close at the end of the extension's lifecycle - in our case, with `ContainerStateExtension`, this means at the end of the test where the parameter is injected.

File: update-tests/src/test/java/io/zeebe/test/ContainerState.java [hunk 5]
```diff
@@ -92,6 +97,9 @@ public void start(final boolean enableDebug) {
     broker =
         new ZeebeContainer("camunda/zeebe:" + brokerVersion)
             .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
+            .withEnv("ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE", "128KB")
+            .withEnv("ZEEBE_BROKER_DATA_USEMMAP", "true")
+            .withEnv("ZEEBE_BROKER_DATA_LOGSEGMENTSIZE", "64MB")
```
Comment (npepinpe) [inline update-tests/src/test/java/io/zeebe/test/ContainerState.java:102]: These settings are known to improve performance - however they're not defaults as changing them would be non-backwards compatible. We can remove them here once that's the case.

File: update-tests/src/test/java/io/zeebe/test/ContainerState.java [hunk 6]
```diff
@@ -45,19 +46,18 @@
...
   private ManagedVolume volume;
+  private Network network;
 
   private String brokerVersion;
   private String gatewayVersion;
 
-  public ContainerState(final Network network) {
-    this.network = network;
+  public ContainerState() {
+    network = Network.SHARED;
```
Comment (MiguelPires) [inline update-tests/src/test/java/io/zeebe/test/ContainerState.java:60]: Why not move the assignment to the declaration and remove the constructor altogether?

**Review by miguelpires — APPROVED** [review]
Looks good. I added one small comment but it shouldn't block the PR

File: update-tests/src/test/java/io/zeebe/test/ContainerState.java (hunk 6 above)
Comment (npepinpe) [reply update-tests/src/test/java/io/zeebe/test/ContainerState.java:60]: Ah, I think I had it final for a bit, and just didn't change it after.

**Pull request merged**